MovingBlocks/Terasology
# fix(json-config-loading): Fix handling invalid AutoConfigs and config writing

### Contains

* fix invalid json config writing
* add logging when config invalid
* use defaults when config invalid
* Fixes #4642 

### How to test

Chapter 1 (fix writing, require default settings):
1. Open TS
2. Open Settings -> autoconfig
3. change one of setting (set non-default)
4. exit game
5. Open TS
6. Open Settings -> autoconfig
7. change same settings to default
8. exit game
9. check config file - content should be `{}` (all defaults)

Chapter 2 (logging and invalid config):
1. make Config file as invalid - to corrupt json structure
2. Open TS
3. Check Logs - you should see error log entry about config loading error
4. Check how renders invalid config in autoconfig - should be default
5. exit game
6. check config file - content should be `{}` (all defaults) if you don't change settings at (4)
